### PR TITLE
Feat: Add Mailcow oauth source

### LINKED ISSUE #2369: Support for Mailcow OAuth
**Is your feature request related to a problem? Please describe.**
I'd like to use [Mailcow](https://github.com/mailcow/mailcow-dockerized) as an authentication source (via "Federation&social login"). However, they don't seem to support OpenID connect, just some generic OAuth which [works with Nextcloud](https://mailcow.github.io/mailcow-dockerized-docs/third_party/third_party-nextcloud/) but not many other services. Because of this, I'd like to use Authentik as a bridge between Mailcow and other services.

**Describe the solution you'd like**
It would be nice if Mailcow could be supported like other special providers.

**Additional context**
Mailcow only supports the scope `profile`.
This is what the oauth settings page in mailcow looks like:
image

### PR DESCRIPTION
**Pull request by dorianim**
# Details
* **Does this resolve an issue?**
Resolves #2369

### New Features
* Adds Mailcow as an OAuth source

### CODE REVIEW

**Review by BeryJu — CHANGES_REQUESTED** [review]
Couple things missing;

- please run `make migrations` to create migrations for the new model
- please run `make` to format everything
- and finally run `make gen-build` to build the new API scheme

**Comment by dorianim** [comment]
make migrations did not work for me, so I ran `manage.py makemigrations`

**Comment by BeryJu** [comment]
Also I assume you tested this locally, could you add some basic tests like https://github.com/goauthentik/authentik/blob/master/authentik/sources/oauth/tests/test_type_discord.py ?

**Comment by dorianim** [comment]
Tests should work, I'll also add some documentation

**Comment by dorianim** [comment]
All done now :)

File: authentik/sources/oauth/tests/test_type_mailcow.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+
+class TestTypeMailcow(TestCase):
+    """OAuth Source tests"""
+
+    def setUp(self):
+        self.source = OAuthSource.objects.create(
+            name="test",
+            slug="test",
+            provider_type="openidconnect",
```
Comment (BeryJu) [inline authentik/sources/oauth/tests/test_type_mailcow.py:28]: `provider_type="mailcow",`

**Comment by dorianim** [comment]
What did I do wrong with the website? I tried to add the mailcow doc to the menu in `sidebarsIntegrations.js` and the build fails.

**Comment by BeryJu** [comment]
image
that's probably why

**Pull request merged**